activesupport/lib/active_support/core_ext/array/access.rb

```diff
@@ -49,6 +49,18 @@ def excluding(*elements)
   end
   alias :without :excluding
 
+  # Calculates an average from the elements.
+  #  [5, 15, 10, nil].average # => 10
+  #
+  # The default average of an empty list is nil. You can override this default:
+  #  [].average(0) # => 0
+  def average(default = nil)
+    arr = self.compact.select{|x| x.is_a? Numeric}
```

Comment:
Hi! Do you need to do a `compact` if you are anyway doing a `select` for `Numeric` elements?

Reply:
Hi! You're right, should I modify and submit a new PR?

Reply:
Just opened a new PR 
https://github.com/rails/rails/pull/44677